libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1175,6 +1172,27 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 				result ^= fromBigEndian<uint64_t>(function.first.ref());
 			define(_memberAccess) << formatNumber(u256{result} << (256 - 32)) << "\n";
 		}
+		else if (member == "min" || member == "max")
+		{
+			MagicType const* arg = dynamic_cast<MagicType const*>(_memberAccess.expression().annotation().type);
+			IntegerType const* integerType = dynamic_cast<IntegerType const*>(arg->typeArgument());
+			u256 min, max;
+			if (integerType->isSigned())
+			{
+				min = s2u(s256(integerType->minValue()));
```

Comment:
Maybe we can have another pair of functions on IntegerType for this to reduce code duplication?